e2e/test/MessageSendTests.cs

```diff
@@ -496,6 +500,79 @@ public async Task X509_DeviceSendSingleMessage_Http()
+        }
+
+        [TestMethod]
+        public async Task MultiplexingTest_Amqps_Ws()
+        {
+            await SendMessagesOverMultiplex(Client.TransportType.Amqp_WebSocket_Only, MultiplexingDeviceCount).ConfigureAwait(true);
+        }
+
+        [TestMethod]
+        public async Task Message_DeviceClientCanBeClosedAndReOpenedSequentially()
+        {
+            foreach (Client.TransportType transportType in Enum.GetValues(typeof(Client.TransportType)))
+            {
+                await DeviceClientCanBeClosedAndReOpenedSequentially(closeAndReopenCount, transportType).ConfigureAwait(true);
+            }
+        }
+
+        [TestMethod]
+        public async Task Message_DeviceClientCanBeClosedAndReOpenedInParallel()
+        {
+            foreach (Client.TransportType transportType in Enum.GetValues(typeof(Client.TransportType)))
+            {
+                await DeviceClientCanBeClosedAndReOpenedInParallel(closeAndReopenCount, transportType).ConfigureAwait(true);
+            }
+        }
+
+        private async Task DeviceClientCanBeClosedAndReOpenedSequentially(int OpenAndCloseCount, Client.TransportType transportType)
+        {
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(DevicePrefix).ConfigureAwait(false);
+
+            for (int attempt = 0; attempt < OpenAndCloseCount; attempt++)
+            {
+                DeviceClient deviceClient = DeviceClient.CreateFromConnectionString(Configuration.IoTHub.ConnectionString, testDevice.Id, transportType);
+                try
+                {
+                    await OpenCloseAndSendMessage(deviceClient).ConfigureAwait(true);
+                }
+                catch (Exception e)
+                {
+                    throw new Exception("Failed to open and close device client using " + transportType + " on attempt " + attempt, e);
+                }
+            }
+        }
+
+        private async Task DeviceClientCanBeClosedAndReOpenedInParallel(int OpenAndCloseCount, Client.TransportType transportType)
+        {
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(DevicePrefix).ConfigureAwait(false);
+
+            for (int attempt = 0; attempt < OpenAndCloseCount; attempt++)
+            {
+                DeviceClient deviceClient = DeviceClient.CreateFromConnectionString(Configuration.IoTHub.ConnectionString, testDevice.Id, transportType);
+                OpenCloseAndSendMessage(deviceClient); //do not await, want this method to continue before the call finishes
+            }
+        }
+
+        private async Task OpenCloseAndSendMessage(DeviceClient deviceClient)
+        {
+            try
+            {
+                await deviceClient.OpenAsync().ConfigureAwait(true);
+                await deviceClient.SendEventAsync(new Client.Message()).ConfigureAwait(true);
+                await deviceClient.CloseAsync().ConfigureAwait(true);
+            }
+            catch (Exception e)
```

Comment:
Never catch exceptions that can't be handled. The test harness will automatically report it.